src/elf/note.rs

```diff
@@ -85,6 +85,27 @@ if_std! {
         }
     }
 
+    /// An iterator over ELF binary notes
+    pub struct NoteIterator<'a> {
+        pub iters: Vec<NoteDataIterator<'a>>,
+        pub index: usize,
+    }
+
+    impl<'a> Iterator for NoteIterator<'a> {
+        type Item = error::Result<Note<'a>>;
+        fn next(&mut self) -> Option<Self::Item> {
+            while self.index < self.iters.len() {
+                if let Some(note_result) = self.iters[self.index].next() {
+                    return Some(note_result);
+                }
+
+                self.index += 1;
```

Comment:
Won't this only return the first element of every iterator (e.g., shouldn't we also exhaust the iterator at `self.index` until it returns `None`, and then update the `self.index` to point to next iterator?)

Reply:
Let me add a test for this like you suggested, in any case. However, I think this should be correct since `NoteIterator::next` will be called multiple times. Only if a `NoteDataIterator` is exhausted, it will hit the "else" case and increment the index. Otherwise, it just returns the found element and tries again in the next invocation. And since a `NoteDataIterator` could be entirely empty, the while loop tries until it finds one that still contains an item or there are none left.

Reply:
Ah derp, you’re totally right! The early return in the loop was mixing me up for some reason

Reply:
No hard feelings if you'd prefer a different coding style :) Just let me know.

Reply:
No I just needed coffee ;P